### PR TITLE
[feature](agg) add aggregate function sum0

### PR DESCRIPTION
**Pull request by xiedeyantu**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xiedeyantu** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+import java.util.List;
+
+/**
+ * AggregateFunction 'sum0'. sum0 returns the sum of the values which go into it like sum.
+ * It differs in that when no non null values are applied zero is returned instead of null.
+ */
+public class Sum0 extends AggregateFunction
+        implements UnaryExpression, AlwaysNotNullable, ExplicitlyCastableSignature, ComputePrecisionForSum,
+        SupportWindowAnalytic, CouldRollUp {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java:52]: do sum0 really support window?

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java:52]: @seawinde PTAL, does sum0 could implement CouldRollUp?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                "can't convert to multi_distinct_sum because there is no distinct args");
+        return new MultiDistinctSum(false, child());
+    }
+
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        DataType argType = child().getDataType();
+        if ((!argType.isNumericType() && !argType.isBooleanType() && !argType.isNullType())
+                || argType.isOnlyMetricType()) {
+            throw new AnalysisException("sum requires a numeric or boolean parameter: " + this.toSql());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java:90]:
```suggestion
            throw new AnalysisException("sum0 requires a numeric or boolean parameter: " + this.toSql());
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+     */
+    public Sum0(boolean distinct, Expression arg) {
+        super("sum0", distinct, arg);
+    }
+
+    public MultiDistinctSum convertToMultiDistinct() {
+        Preconditions.checkArgument(distinct,
+                "can't convert to multi_distinct_sum because there is no distinct args");
+        return new MultiDistinctSum(false, child());
+    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java:83]: sum0 should convert to MultiDistinctSum0?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java (hunk 1 above)
Comment (xiedeyantu) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Sum0.java:52]: > @seawinde PTAL, does sum0 could implement CouldRollUp?

keep the behavior to sum?

**Comment by xiedeyantu** [comment]
run buildall

**Comment by xiedeyantu** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by 924060929** [comment]
add regression test please

**Comment by xiedeyantu** [comment]
run buildall

**Comment by xiedeyantu** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**